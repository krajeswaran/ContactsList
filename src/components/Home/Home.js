import React, {useState, useEffect} from 'react';
import {
  SafeAreaView,
  Text,
  StatusBar,
  TouchableOpacity,
  View,
  StyleSheet
} from 'react-native';

const Home = ({route,navigation}) =>  {
  const [selected_contact,setSelectedContact] = useState(null);

  //Checking if any selected contact is present (It will be found after the selection is made from the list component)
  useEffect(() => {
        if(route.params && route.params.confirmed_select){
            setSelectedContact(route.params.confirmed_select)
        }
  }, [route.params])

  return (
    <>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView style={styles.container}>
        <TouchableOpacity onPress={() => navigation.navigate('Contacts')}>
            <View style={styles.select_button}>
                <Text style={styles.button_text}>
                    Select a Contact
                </Text>
            </View>
        </TouchableOpacity>

        {/*Displaying selected contact*/}
        {selected_contact && <View style={styles.selected_container}>
            <Text style={styles.select_contact_header}>
                Selected Contact
            </Text>
            <View style={styles.selected_content}>
                <Text style={styles.selected_title}>
                    {selected_contact.title} ::
                </Text>
                <Text style={styles.selected_number}>
                    {" " + selected_contact.phone_number.number}
                </Text>
            </View>
        </View>}
      </SafeAreaView>
    </>
  );
};

const styles = StyleSheet.create({
    container:{
        flex:1,
        alignContent:'center',
        justifyContent:'center'
    },
    select_button:{
        backgroundColor:'white',
        padding:10,
        display:'flex',
        justifyContent:'center',
        flexDirection:'row'
    },
    button_text:{
        fontSize:16,
        fontWeight:'bold'
    },
    selected_container:{
        display:'flex',
        justifyContent:'center',
        alignItems:'center',
        flexDirection:'column',
        flex:2
    },
    selected_content:{
        display:'flex',
        justifyContent:'center',
        flexDirection:'row',
        marginTop:10
    },
    selected_title:{
        fontWeight:'bold',
        color:'blue',
        fontSize:15
    },
    selected_number:{
        fontWeight:'bold',
        color:'red',
        fontSize:15
    },
    select_contact_header:{
        fontSize:16
    }
})

export default Home;