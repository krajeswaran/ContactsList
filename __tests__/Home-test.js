import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/components/Home/Home';

jest.setTimeout(15000);

it('renders correctly', () => {
  const params = {confirmed_select:{title:'abcd',phone_number:{number:123456789,label:'work'}}}
  const tree = renderer.create(<Home route={params}/>).toJSON();
  expect(tree).toMatchSnapshot();
});