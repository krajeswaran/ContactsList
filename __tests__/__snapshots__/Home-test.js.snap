// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignContent": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Select a Contact
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
